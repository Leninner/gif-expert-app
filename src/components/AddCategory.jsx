import React, { useState } from 'react';
import { Button, TextField } from '@mui/material';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Handle submit');

    if (inputValue.trim().length > 2) {
      setCategories(inputValue);
      setInputValue('');
    }
  };

  return (
    <form className='divito' onSubmit={handleSubmit}>
      <TextField label='Category' focused onChange={handleInputChange} value={inputValue} />
      <p>{inputValue}</p>
      <Button variant='contained' type='submit'>
        Add Category
      </Button>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
