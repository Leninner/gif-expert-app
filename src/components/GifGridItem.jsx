import React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';

export const GifGridItem = ({ title, url }) => {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia image={url} component='img' height='140' />
      <CardContent>
        <Typography gutterBottom variant='h5' component='h2'>
          {title}
        </Typography>
      </CardContent>
    </Card>
  );
};
