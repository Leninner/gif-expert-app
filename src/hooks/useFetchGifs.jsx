import { useState, useEffect } from 'react';
import { getData } from '../helpers/getData';
import PropTypes from 'prop-types';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({ data: [], loading: true });

  useEffect(() => {
    getData(category).then((imgs) => {
      setState({ loading: false, data: imgs });
    });
  }, [category]);

  return state;
};

useFetchGifs.propTypes = {
  category: PropTypes.string.isRequired,
};
