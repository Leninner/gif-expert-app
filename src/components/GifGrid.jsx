import React from 'react';
import { GifGridItem } from './GifGridItem';
import Grid from '@mui/material/Grid';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h1>{category}</h1>
      <Grid container spacing={{ xs: 1, md: 3 }} columns={{ xs: 2, sm: 8, md: 12 }} sx={{ flexGrow: 0 }}>
        {loading && <h2>Loading...</h2>}
        {images.map((img) => (
          <Grid item xs={2} sm={4} md={4} key={img.id}>
            <GifGridItem {...img} />
          </Grid>
        ))}
      </Grid>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
