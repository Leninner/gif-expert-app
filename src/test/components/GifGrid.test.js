import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Vamos a tener una buena prueba', () => {
  const category = 'One Punch';
  const gifs = [{ id: 'ABC', url: 'https://localhost/one-punch.gif', title: 'One Punch' }];

  useFetchGifs.mockReturnValue({ data: gifs, loading: true });

  const wrapper = shallow(<GifGrid category={category} />);

  test('Debe coincidir con el snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar las imÃ¡genes cuando se va cargando', () => {});
});
