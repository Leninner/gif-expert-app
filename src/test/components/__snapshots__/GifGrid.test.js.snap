// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vamos a tener una buena prueba Debe coincidir con el snapshot 1`] = `
<Fragment>
  <h1>
    One Punch
  </h1>
  <ForwardRef(Grid)
    columns={
      Object {
        "md": 12,
        "sm": 8,
        "xs": 2,
      }
    }
    container={true}
    spacing={
      Object {
        "md": 3,
        "xs": 1,
      }
    }
    sx={
      Object {
        "flexGrow": 0,
      }
    }
  >
    <h2>
      Loading...
    </h2>
    <ForwardRef(Grid)
      item={true}
      key="ABC"
      md={4}
      sm={4}
      xs={2}
    >
      <GifGridItem
        id="ABC"
        title="One Punch"
        url="https://localhost/one-punch.gif"
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</Fragment>
`;
